### PR TITLE
SUSI now stops Music while playing Second query

### LINKED ISSUE #298: To pause the music while SUSI gives out second query
**Describe the bug**
Right now when SUSI processes second query, music doesn't stop and the query is spoken.

**To Reproduce**
1. Ask SUSI to play Music 
2. Ask another query

**Expected behavior**
To pause and play the music again

**Additional context**
I would like to work on it

### PR DESCRIPTION
**Pull request by stealthanthrax**
Fixes #298 

SUSI can now process the second query while playing Music and will pause the music throughout the process

### CODE REVIEW

File: main/states/busy_state.py [hunk 1]
```diff
@@ -13,16 +13,26 @@ class BusyState(State):
...
-        subprocess.call(['play', str(self.components.config['detection_bell_sound'])])  # nosec #pylint-disable type: ignore
-        self.transition(self.allowedStateTransitions.get('recognizing'))
-        os.system('kill CONT ' + str(self.video_pid))  # nosec #pylint-disable type: ignore
+        if 'video_pid' in self.__dict__:
+            os.system('kill -STOP ' + str(self.video_pid))  # nosec #pylint-disable type: ignore
+            lights.wakeup()
+            subprocess.Popen(['play', str(self.components.config['detection_bell_sound'])])  # nosec #pylint-disable type: ignore
+            lights.off()
+            self.transition(self.allowedStateTransitions.get('recognizing'))
+            os.system('kill -CONT ' + str(self.video_pid))  # nosec #pylint-disable type: ignore
```
Comment (stealthanthrax) [inline main/states/busy_state.py]: I tried using `subprocess` instead of `os` module but  'kill -STOP/-CONT' are not working in `subprocess` module

**Comment by stealthanthrax** [comment]
@prateekiiest , have a look too

File: main/states/busy_state.py [hunk 2]
```diff
@@ -13,16 +13,26 @@ class BusyState(State):
...
         """This callback is fired when a Hotword Detector detects a hotword.
         :return: None
         """
         # subprocess.call(['killall', 'play'])
         # subprocess.call(['killall', 'mpv']
-        os.system('kill STOP ' + str(self.video_pid))  # nosec #pylint-disable type: ignore
-        subprocess.call(['play', str(self.components.config['detection_bell_sound'])])  # nosec #pylint-disable type: ignore
-        self.transition(self.allowedStateTransitions.get('recognizing'))
-        os.system('kill CONT ' + str(self.video_pid))  # nosec #pylint-disable type: ignore
+        if 'video_pid' in self.__dict__:
```
Comment (hongquan) [inline main/states/busy_state.py]: Better:

```py
if getattr(self, `video_pid`):
```

Avoiding abusing `__` functions.

Comment (stealthanthrax) [reply main/states/busy_state.py]: @hongquan , on using this approach , I am getting an error `'BusyState' object has no attribute 'audio_pid'`
>if getattr(self, `video_pid`):

**Comment by stealthanthrax** [comment]
@hongquan , what I meant was that `subprocess.call['kill','-CONT']` doesn't work very well. I have implemented it in a different way. Please have  a look.
>I tried using subprocess instead of os module but 'kill -STOP/-CONT' are not working in subprocess module

File: main/states/busy_state.py (hunk 2 above)
Comment (hongquan) [reply main/states/busy_state.py]: Show me the code and the log of that error.

You haven't assigned `audio_pid`, so of course `no attribute 'audio_pid'`.

**Comment by hongquan** [comment]
Don't name `video_pid` when it is the value returned from `subprocess.Popen()`. It is not a PID.

File: main/states/busy_state.py (hunk 2 above)
Comment (hongquan) [reply main/states/busy_state.py]: Sorry, I remember wrong the function. Should be `hasattr`, not `getattr`.

**Comment by stealthanthrax** [comment]
@hongquan , updated. Please have a look

**Pull request merged**